currencycloud/__init__.py

```diff
@@ -14,55 +15,61 @@
 token = None
 
 # global session management
-__session = None
+thread_storage = threading.local()
 
 
 def session(authenticate=True):
-    global __session
+    __ensure_thread_session()
+    if not thread_storage.session:
+        thread_storage.session = Session(
+                                    environment,
+                                    login_id,
+                                    api_key,
+                                    token,
+                                    authenticate=authenticate)
+    session = thread_storage.session
 
-    if not __session:
-        __session = Session(
-            environment,
-            login_id,
-            api_key,
-            token,
-            authenticate=authenticate)
+    if not session.authenticated and authenticate is True:
+        session.authenticate()
 
-    if not __session.authenticated and authenticate is True:
-        __session.authenticate()
-
-    return __session
+    return session
 
 
 def close_session():
-    global __session
-    if __session:
-        __session.close()
-        __session = None
+    __ensure_thread_session()
+
+    if thread_storage.session:
+        thread_storage.session.close()
+        thread_storage.session = None
     return True
 
 
 def reset_session():
-    global __session
-    __session = None
+    __ensure_thread_session()
+    thread_storage.session = None
     token = None
 
 
 @contextmanager
 def on_behalf_of(contact_id):
-    global __session
+    __ensure_thread_session
```

Comment:
:grimacing: I did miss that but after wondering why it was still working without that call, I realised that the `session` method would set it up. I've removed it in a subsequent commit.